pallets/sequencer-staking/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+	}
+
+	#[pallet::error]
+	/// Errors
+	pub enum Error<T> {
+		OperationFailed,
+		MathOverflow
+	}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type Currency: ReservableCurrency<Self::AccountId>;
+		#[pallet::constant]
+		type MinimumSequencers: Get<u32>;
+		type RolldownProvider: RolldownProviderTrait<Self::AccountId>;
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::DbWeight::get().reads_writes(4, 4).saturating_add(Weight::from_parts(40_000_000, 0)))]
+		pub fn provide_sequencer_stake(origin: OriginFor<T>, stake_amount: BalanceOf<T>) -> DispatchResultWithPostInfo {
+			
+			let sender = ensure_signed(origin)?;
+
+			<SequencerStake<T>>::try_mutate(&sender, |stake| -> DispatchResult {
+				let previous_stake = *stake;
+				*stake = stake.checked_add(&stake_amount).ok_or(Error::<T>::MathOverflow)?;
+				if previous_stake < MinimalStakeAmount::<T>::get() && *stake >= MinimalStakeAmount::<T>::get(){
+					T::RolldownProvider::new_sequencer_active(sender.clone());
+				}
+				Ok(())
+			})?;
+
+			T::Currency::reserve(&sender, stake_amount)?;
+
+			Ok(().into())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::DbWeight::get().reads_writes(2, 2).saturating_add(Weight::from_parts(40_000_000, 0)))]
+		pub fn set_sequencer_configuration(origin: OriginFor<T>, minimal_stake_amount: BalanceOf<T>, slash_fine_amount: BalanceOf<T>) -> DispatchResultWithPostInfo {
+			
+			let _ = ensure_root(origin)?;
+			
+			<MinimalStakeAmount<T>>::put(minimal_stake_amount);
+			<SlashFineAmount<T>>::put(slash_fine_amount);
+
+			Ok(().into())
+
+		}
+	}
+}
+
+impl<T: Config> SequencerStakingProviderTrait<AccountIdOf<T>, BalanceOf<T>> for Pallet<T> {
+
+	fn is_active_sequencer(sequencer: AccountIdOf<T>) -> bool{
+
+		<SequencerStake<T>>::get(&sequencer) >= MinimalStakeAmount::<T>::get()
+
+	}
+
+	fn slash_sequencer(sequencer: AccountIdOf<T>) -> DispatchResult{
+
+		<SequencerStake<T>>::try_mutate(&sequencer, |stake| -> DispatchResult {
+			let slash_fine_amount = SlashFineAmount::<T>::get();
+			let slash_fine_amount_actual = (*stake).min(slash_fine_amount);
+			*stake = stake.saturating_sub(slash_fine_amount_actual);
+			let _ = T::Currency::slash_reserved(&sequencer, slash_fine_amount_actual);
+			Ok(())
+		})?;
+		
+		Ok(().into())
+	}
+
+	fn process_potential_authors(authors: Vec<(AccountIdOf<T>, BalanceOf<T>)>) -> Option<Vec<(AccountIdOf<T>, BalanceOf<T>)>> {
+		let minimal_stake_amount = MinimalStakeAmount::<T>::get();
+		let filtered_authors: Vec<_> = authors.into_iter().filter( |a| <SequencerStake<T>>::get(&a.0) >= minimal_stake_amount).collect::<>();
+		if filtered_authors.len() as u32 >= T::MinimumSequencers::get(){
+			Some(filtered_authors)
+		} else {
+			None
```

Comment:
What I do understand here is that for now the minimal sequencer stake is mandatory for any collator. But what I do understand from the doc, is that we should not filter  by this sequencer stake unless we have two fo them? 
Is not quite risky return no candidates? whill this halt the chain?

Reply:
In the parachain-staking pallet you can see that if this function returns none - the filter is not applied.

Reply:
niice!